File: perf/matmul.c
#include<stdio.h>
#include<math.h>
#define N 100

void matrixprint(double A[N][N])
{
  int i,j;
  for(i=0;i<N;i++){
    for(j=0;j<N;j++)
      printf("\t%f,",A[i][j]);
    printf("\n");
  }
}

void matrixrandom(double A[N][N])
{
   int i,j;
   for(i=0;i<N;i++)
     for(j=0;j<N;j++){
       A[i][j]=sin(sin((double)(i)+1.1)/cos((double)(j)-1.1));
     }
}

void matrixmul(double C[N][N],double A[N][N],double B[N][N])
{
  int i,j,s;
  for(i=0;i<N;i++)
    for(j=0;j<N;j++){
      C[i][j]=0.;
      for(s=0;s<N;s++)
        C[i][j]+=A[i][s]*B[s][j];
    }
}

int main()
{
  double A[N][N];
  double C[N][N];
  int i;
  matrixrandom(A);
  printf("%f\n",A[10][10]);
  /*matrixprint(A);*/
  for(i=0;i<1000;i++)
    matrixmul(C,A,A);
  return 0;
}

File: perf/matmul.lua
matrix={}

matrix.new=function(dim)
  local A={}
  for i=1,dim do table.insert(A,{}) end
  return A
end

matrix.random=function(dim)
  local A=matrix.new(dim)
  local random=math.random
  for i=1,dim do
    for j=1,dim do
      A[i][j]=math.sin(math.sin(i+1.1)/math.cos(j-1.1))
    end
  end
  return A
end

matrix.mul=function(A,B)
  local dim=#A
  local C=matrix.new(dim)
  for i=1,dim do
    for j=1,dim do
      C[i][j]=0
      for s=1,dim do
        C[i][j]=C[i][j]+A[i][s]*B[s][j]
      end
    end
  end
  return C
end

t=matrix.random(100)
print(t[10][10])
for i=1,1000 do
  matrix.mul(t,t)
end
